src/core/createInstance.js

```diff
@@ -43,48 +42,59 @@ export default (
     }
     const config = createConfig(options);
 
-    try {
-      componentRegistry = initializeComponents(config);
-    } catch (e) {
-      configurationFailed = true;
-      throw e;
-    }
+    return initializeComponents(config);
```

Comment:
Notice how the configure command now returns the promise from `initializeComponents`? That's a key piece to the error handling.